### PR TITLE
expandConfig to accept parameter optionalMatch

### PR DESCRIPTION
**Pull request by dmk255**
In response to #180

### CODE REVIEW

File: src/main/java/apoc/path/PathExplorer.java [hunk 1]
```diff
@@ -21,6 +22,7 @@
 	private static final String VERSION = "0.5";
 	public static final Uniqueness UNIQUENESS = Uniqueness.RELATIONSHIP_PATH;
 	public static final boolean BFS = true;
+	public static final boolean OPTIONALMATCH = false;
```
Comment (jexp) [inline src/main/java/apoc/path/PathExplorer.java]: not used and not needed

File: src/main/java/apoc/path/PathExplorer.java [hunk 2]
```diff
@@ -95,16 +98,30 @@ private Uniqueness getUniqueness(String uniqueness) {
...
 		// LabelFilter
 		// -|Label|:Label|:Label excluded label list
 		// +:Label or :Label include labels
 
-		Traverser traverser = traverse(db.traversalDescription(), startNodes, pathFilter, labelFilter, minLevel, maxLevel, uniqueness,bfs);
-		return traverser.stream().map( PathResult::new );
+		Traverser traverser = traverse(db.traversalDescription(), startNodes, pathFilter, labelFilter, minLevel, maxLevel, uniqueness,bfs, optionalMatch);
+	    
+		Stream<PathResult> returnStream = traverser.stream().map( PathResult::new );
+		if (optionalMatch) {
```
Comment (jexp) [inline src/main/java/apoc/path/PathExplorer.java]: can we simplify this to test on traverser `if (!optional || traverser.hasNext())` and then return the regular stream?

File: src/main/java/apoc/path/PathExplorer.java [hunk 3]
```diff
@@ -95,16 +98,30 @@ private Uniqueness getUniqueness(String uniqueness) {
...
+		    Iterator<PathResult> iterator = returnStream.iterator();
+		    if (iterator.hasNext()) {
+		    	returnStream = StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, 0), false);
+		    } else {
+		    	List<PathResult> originalNodesAsPaths = new ArrayList<>();
```
Comment (jexp) [inline src/main/java/apoc/path/PathExplorer.java]: `return startNodes.stream().map( n -> Paths.singleNodePath(n)).map(PathResult::new)`

**Comment by jexp** [comment]
Please rebase to master not 3.1 branch.

**Comment by jexp** [comment]
ping @dmk255

**Comment by jexp** [comment]
ping @dmk255

**Comment by jexp** [comment]
@InverseFalcon could you have a look at this? as this is now your familar ground? :)

**Comment by dmk255** [comment]
Hey @jexp I'll make the changes to this on monday if that works. I missed the previous notifications to this thread.

**Comment by jexp** [comment]
@InverseFalcon I think we can close this as fixed, or?

**Comment by InverseFalcon** [comment]
Yes, #422 fulfilled optional functionality. Close implementations!

**Pull request closed without merging**